Grouping the mobilenetv2_coco models

sayakpaul opened the pull request:
@maringeo this PR groups together the [COCO models](https://tfhub.dev/s?publisher=sayakpaul&q=deeplabv3-mobilenetv2) and contains metadata populated TFLite models. 

Cc: @khanhlvg

maringeo commented on tfhub_dev/assets/sayakpaul/deeplabv3-mobilenetv2/default/lite/1.md:
```diff
@@ -1,19 +0,0 @@
-# Lite sayakpaul/deeplabv3-mobilenetv2/1/default/1
```
The corresponding placeholder MD for this Lite MD files is not modified. Is this intended?

maringeo commented on tfhub_dev/assets/sayakpaul/mobilenetv2-coco/int8/1.md:
```diff
@@ -0,0 +1,26 @@
+# Lite sayakpaul/mobilenetv2-coco/int8/1
```
Is the goal of the PR to reorganize models into different directories? If yes, we should keep the first line of each MD file (model identifier) unchanged.

Changing the model identifier (also known as handle) from `sayakpaul/deeplabv3-mobilenetv2-int8/1/default/1` to `sayakpaul/mobilenetv2-coco/int8/1` will break all existing TF Hub users that load the model via url, for example:

`hub.load("https://tfhub.dev/sayakpaul/lite-model/deeplabv3-mobilenetv2-int8/1/default/1")`

maringeo commented on tfhub_dev/assets/sayakpaul/deeplabv3-mobilenetv2-float16/1.md:
```diff
@@ -1,9 +0,0 @@
-# Placeholder sayakpaul/deeplabv3-mobilenetv2-float16/1
```
The placeholder MD files have to moved together with the Lite MD files.

sayakpaul commented:
@maringeo the purpose is to group together the models belonging to the same checkpoints and also update those models with metadata population. I will work on the PR to address your comments. Thank you.

maringeo commented:
> @maringeo the purpose is to group together the models belonging to the same checkpoints and also update those models with metadata population. I will work on the PR to address your comments. Thank you.

Thanks @sayakpaul for the explanation, I have two follow-up questions:
- do you want to group these models in a collection on tfhub.dev (similar to https://tfhub.dev/google/collections/efficientnet/1) or is this grouping purely affecting the directory structure of the MD files in the GitHub repo?
- if the underlying .tflite files are changed to contain metadata, we should publish these as version 2 of the current models, since the TF Hub asset files (.tflite, .tfjs, saved_model.pb) are immutable (for backwards compatibility reasons)

sayakpaul commented:
@maringeo the grouping similar to this: https://tfhub.dev/google/magenta/arbitrary-image-stylization-v1-256/2. Would you suggest to keep the v1 models (i.e. the nonpopulated ones) as they are?

maringeo commented:
> @maringeo the grouping similar to this: https://tfhub.dev/google/magenta/arbitrary-image-stylization-v1-256/2. Would you suggest to keep the v1 models (i.e. the nonpopulated ones) as they are?

All clear, thanks! In that case, the Lite MD files should reference the same Placeholder MD as parent-model. I see that you've already done this in the PR.

Regarding keeping the v1 models - I think we should not change anything in the v1 MD content. File-organization wise, I think it is a good idea to keep the v1 and v2 model MDs under the same parent directory, so if you add the v2 files in a separate directory, could you also move the v1 versions there?

One final suggestion - it's often a good idea to outline in the MD description what changed between model versions. This makes it easier for TF Hub users to pick the right version for their use case. Can you add to the v2 MDs that the difference between v1 and v2 models is the extra TFLite metadata?

sayakpaul commented:
> One final suggestion - it's often a good idea to outline in the MD description what changed between model versions. This makes it easier for TF Hub users to pick the right version for their use case. Can you add to the v2 MDs that the difference between v1 and v2 models is the extra TFLite metadata?

@maringeo I think I have made a note about that already. If that is not clear please let me know. 

Another issue when I was creating the v2 models I did not know about the grouping capability. So, I created all the models separately. For example, [this one](https://tfhub.dev/sayakpaul/lite-model/deeplabv3-mobilenetv2/1/default/1) and [this one ](https://tfhub.dev/sayakpaul/lite-model/deeplabv3-mobilenetv2_dm05-int8/1/default/1) are basically exported from the same checkpoint. This PR groups them under one roof while only keeping their v2 variant.

maringeo commented:
Thanks, I saw the note. Also, just fyi, since today, tfhub.dev renders TF Lite metadata within the model page. You can see an example at https://tfhub.dev/sayakpaul/lite-model/arbitrary-image-stylization-inceptionv3/dr/predict/1

Overall the plan sounds good. I think that we should also modify the v1 Lite MDs to point to the new Placeholder MD files that is used for the grouping. In this way, https://tfhub.dev/sayakpaul/lite-model/deeplabv3-mobilenetv2-int8/1/default/1 would still resolve correctly. What do you think?

sayakpaul commented:
Fair enough. That sounds pretty. The metadata preview is so cool. @maringeo

sayakpaul commented:
@maringeo I have addressed your feedback. Added the following note:

> - If you are looking for metadata populated models you should refer [here](https://tfhub.dev/sayakpaul/lite-model/mobilenetv2-coco/dr/1). 

I hope the URL (when the model would be up on Hub) is correct.

maringeo commented on tfhub_dev/assets/sayakpaul/mobilenetv2-coco/1.md:
```diff
@@ -0,0 +1,9 @@
+<!-- network-architecture: DeepLab (mobilenetv2_coco_voc_trainval) -->
+<!-- dataset: PASCAL VOC 2012 -->
+<!-- fine-tunable: false -->
+<!-- language: en -->
```
The language metadata property is not applicable to image models, so we can remove this (unless I'm missing something).

sayakpaul replied:
@maringeo fixed it.

maringeo commented on tfhub_dev/assets/sayakpaul/mobilenetv2-coco/1.md:
```diff
@@ -0,0 +1,8 @@
+# Placeholder sayakpaul/deeplabv3-mobilenetv2/1
```
Sorry, missed this earlier - should this be:
`# Placeholder sayakpaul/mobilenetv2-coco/1`

sayakpaul replied:
Norry worries @maringeo. I should have been more careful. Thanks for always being so supportive. I just corrected it.

maringeo commented:
Hi @sayakpaul , the models are live on https://tfhub.dev/sayakpaul/lite-model/mobilenetv2-coco/dr/1 (also the TF Lite metadata renders correctly at the bottom of the model pages).

This PR has been merged in 64d8c58b52e329f8d5dbd0ee4dd13848779d8cc2.

sayakpaul commented:
@khanhlvg FYI.
